src/runtime/providers/directus.ts

```diff
@@ -0,0 +1,43 @@
+import { joinURL } from 'ufo'
+import type { ProviderGetImage } from '../../types'
+import { createOperationsGenerator } from '#image'
+
+export const operationsGenerator = createOperationsGenerator({
+  keyMap: {
+    key: 'key',
+    width: 'width',
+    height: 'height',
+    format: 'format',
+    quality: 'quality',
+    fit: 'fit',
+    withoutEnlargement: 'withoutEnlargement',
+    transforms: 'transforms'
+  },
+  valueMap: {
+    fit: {
+      cover: 'cover',
+      contain: 'contain',
+      inside: 'inside',
+      outside: 'outside'
+    },
+    format: {
+      jpg: 'jpg',
+      png: 'png',
+      webp: 'webp',
+      tiff: 'tiff'
+    },
+    withoutEnlargement: {
+      true: 'true',
+      false: 'false'
+    }
+  },
+  joinWith: '&',
+  formatter: (key, value) => `${key}=${value}`
+})
+
+export const getImage: ProviderGetImage = (src, { modifiers = {}, baseURL = 'http://localhost:8055/assets' } = {}) => {
```

Comment:
I think it makes sense for a user to have to provide this baseURL in the config - let's not hard-code it in the provider.

Reply:
True, I did it when I was still mainly using [intevel/nuxt-directus](https://github.com/intevel/nuxt-directus)

Reply:
Understood. I think we should update that one as well for the same reason.

Reply:
What about:

```ts
export const getImage: ProviderGetImage = (src, { modifiers = {}, baseURL } = {}) => {
  const operations = operationsGenerator(modifiers)
  if (baseURL && !baseURL.endsWith('/assets')) {
    baseURL += '/assets'
  }
  return {
    url: joinURL(baseURL || 'http://localhost:8055/assets', src + (operations ? ('?' + operations) : ''))
  }
}
```

This way we both handle defined baseURLs (even if the developer forgets to add `/assets`) but still fallback to a default one?